cli/src/auth.ts

```diff
@@ -1,140 +1,202 @@
-    )
-    .option("-r, --rpc <string>", "The endpoint to connect to.")
-    .option("-l, --log-level <string>", "log level", setLogLevel)
-    .option("-a, --address <string>", "The address of the ruleset.")
-    .action(async (directory, cmd) => {
-        const { keypair, env, rpc, address } = cmd.opts();
-        const connection = new Connection(rpc, "finalized");
-
-        let rulesetPDA = new PublicKey(address);
-        let rulesetData = await connection.getAccountInfo(rulesetPDA);
-        let data = rulesetData?.data as Buffer;
-        let ruleset = getLatestRuleSet(data);
-        let rulesetDecoded = JSON.parse(ruleset);
-        console.log(JSON.stringify(rulesetDecoded, pubkey_replacer, 2));
-    });
+  .command('print')
+  .option('-e, --env <string>', 'Solana cluster env name', 'devnet')
+  .option('-r, --rpc <string>', 'The endpoint to connect to')
+  .option('-l, --log-level <string>', 'Log level', setLogLevel)
+  .option('-a, --address <string>', 'The address of the rule set')
+  .option('-o, --output <file>', 'The file to save the output to')
+  .action(async (_directory, cmd) => {
+    let { env, rpc, address, output } = cmd.opts();
+
+    rpc = rpc ?? default_env(env);
+    const connection = new Connection(rpc, 'finalized');
+
+    console.log(`🔎 Retrieving latest rule set revision for '${address}'`);
 
+    let rulesetPDA = new PublicKey(address);
+    let rulesetData = await connection.getAccountInfo(rulesetPDA);
+    let data = rulesetData?.data as Buffer;
+    let ruleset = getLatestRuleSetRevision(data);
+
+    if (output) {
+      console.log('   + writing revision to file');
+      fs.writeFileSync(output, JSON.stringify(ruleset, pubkey_replacer, 2));
+    } else {
+      console.log('\n');
+      console.log(colorizeJson(JSON.stringify(ruleset, pubkey_replacer, 2)));
+    }
+
+    console.log(`\n✅ Revision retrieved.`);
+  });
+
+program.version('1.0.0').description('CLI for managing RuleSet revisions.').parse(process.argv);
+
+//-----------//
+// Helpers   //
+//-----------//
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 function setLogLevel(value, prev) {
-    if (value === undefined || value === null) {
-        return;
-    }
-    log.info("setting the log value to: " + value);
-    log.setLevel(value);
+  if (value === undefined || value === null) {
+    return;
+  }
+  log.info('setting the log value to: ' + value);
+  log.setLevel(value);
 }
 
 function loadKeypair(keypairPath) {
-    const decodedKey = new Uint8Array(
-        JSON.parse(
-            fs.readFileSync(keypairPath).toString()
-        ));
+  const decodedKey = new Uint8Array(JSON.parse(fs.readFileSync(keypairPath).toString()));
 
-    return Keypair.fromSecretKey(decodedKey);
+  return Keypair.fromSecretKey(decodedKey);
 }
 
-program
-    .version("0.0.1")
-    .description("CLI for controlling and managing RuleSets.")
-    .parse(process.argv);
-
 function pubkey_replacer(key, value) {
```

Comment:
Do we still want to use that parser when displaying the JSON instead of returning the JSON as it really is? I can imagine people copy/pasting the output into a JSON file to use as input on a new revision and them not understanding why it doesn't work (because what should be an array of number is now a base58 string). I think it's probably best to output the "real" JSON to avoid confusions.

Additionally, 32-bytes arrays don't only represent public keys in v1. They can also represent Merkle Roots which will be confusing when they are displayed as base58 public keys.

Reply:
Good point!